oak-store-spi/src/main/java/org/apache/jackrabbit/oak/spi/state/ReadyOnlyBuilderUnsupportedOperationException.java

```diff
@@ -0,0 +1,12 @@
+package org.apache.jackrabbit.oak.spi.state;
+
+/**
+ * Indicates that a modification operation was tried to execute on a read-only builder.
+ * It should be used instead of throwing plain UnsupportedOperationExceptions in that situation.
+ */
+public class ReadyOnlyBuilderUnsupportedOperationException extends UnsupportedOperationException {
```

Comment:
Maybe `ReadyOnlyBuilderException` would be sufficiently clear and a bit shorter than `ReadyOnlyBuilderUnsupportedOperationException`

Reply:
done